File: Web/ClientApp/src/app/_models/gameSetup.ts
import { CardValue, GameType, PlayersSetup } from './enums';

export interface GameSetup {
  password: string;
  bannedCards: CardValue[];
  numberOfStandardDecks: number;
  roundsToWin: number;
  gameType: GameType;
  playersSetup: PlayersSetup;
  drawFourDrawTwoShouldSkipTurn: boolean;
  reverseShouldSkipTurnInTwoPlayers: boolean;
  matchingCardStealsTurn:boolean;
  wildCardCanBePlayedOnlyIfNoOtherOptions:boolean;
  maxNumberOfPlayers: number;
  canSeeTeammatesHandInTeamGame: boolean;
  drawAutoPlay: boolean;
  spectatorsCanViewHands: boolean;
  limitColorChangingCards: boolean;
}

File: EntityObjects/GameSetup.cs
using System.Collections.Generic;
using Common.Enums;

namespace EntityObjects
{
    public class GameSetup
    {
        public string Password { get; set; }
        public List<CardValue> BannedCards { get; set; }
        public int RoundsToWin { get; set; }
        public GameType GameType { get; set; }
        public PlayersSetup PlayersSetup { get; set; }
        public bool DrawFourDrawTwoShouldSkipTurn { get; set; }
        public bool ReverseShouldSkipTurnInTwoPlayers { get; set; }
        public bool MatchingCardStealsTurn { get; set; }
        public bool WildCardCanBePlayedOnlyIfNoOtherOptions { get; set; }
        public int MaxNumberOfPlayers { get; set; }
        public bool CanSeeTeammatesHandInTeamGame { get; set; }
    }
}